packages/modular-scripts/src/check.ts

```diff
@@ -1,14 +1,26 @@
+import * as path from 'path';
 import { getWorkspaceInfo, WorkSpaceRecord } from './utils/getWorkspaceInfo';
-import { getModularType, isValidModularType } from './utils/isModularType';
+import {
+  getModularType,
+  isValidModularType,
+  isValidModularRootPackageJson,
+} from './utils/isModularType';
 import * as logger from './utils/logger';
+import getModularRoot from './utils/getModularRoot';
 
 export async function check(): Promise<void> {
   // ensure that workspaces are setup correctly with yarn§
   // init is a special case where we don't already need to be in a modular repository
   // in this case there's no use checking the workspaces yet because we're setting
   // up a new folder
   const workspace = await getWorkspaceInfo();
+  const modularRoot = getModularRoot();
 
+  if (!isValidModularRootPackageJson(modularRoot)) {
+    throw new Error(
```

Comment:
This isn't really a super helpful error - I'll make a PR to make this explicit as to what the real error is.

Reply:
If you want to create an issue to have the method itself throw the error, I can raise the PR